src/compiler/emitter.ts

```diff
@@ -1905,11 +1903,11 @@ namespace ts {
             emitModifiers(node, node.modifiers);
             write("enum ");
             emit(node.name);
-            pushNameGenerationScope();
+            pushNameGenerationScope(node);

```

Comment:
I suppose enums aren't really a place you can inject temporary variables either, so they probably shouldn't be a name-generation scope (even though that only really matters for javavascript output and enums are typescript only).